Zend/zend.c

```diff
@@ -1892,3 +1860,29 @@ ZEND_API void zend_map_ptr_extend(size_t last)
 		CG(map_ptr_last) = last;
 	}
 }
+
+ZEND_API void zend_alloc_ce_cache(zend_string *type_name)
+{
+	if (ZSTR_HAS_CE_CACHE(type_name) || !ZSTR_IS_INTERNED(type_name)) {
+		return;
+	}
+
+	if ((GC_FLAGS(type_name) & IS_STR_PERMANENT) && startup_done) {
+		/* Don't allocate slot on permanent interned string outside module startup.
+		 * The cache slot would no longer be valid on the next request. */
+		return;
+	}
+
+	if (zend_string_equals_literal_ci(type_name, "self")
+			|| zend_string_equals_literal_ci(type_name, "parent")) {
```

Comment:
It may make sense to add ``self`` and ``parent`` into ZEND_KNOWN_STRINGS and compare just pointers.